File: samples/antlr/TestLexer.rb
require "rjava"

module TestLexerImports
  include Object::Java::Lang
  include Object::Org::Antlr::Runtime
  include_const Object::Java::Util, :Stack
  include_const Object::Java::Util, :List
  include_const Object::Java::Util, :ArrayList
end

class TestLexer < TestLexerImports.const_get :Lexer
  include TestLexerImports
  
  lazy_constants[:EOF] = lambda { EOF = -1 }
  lazy_constants[:T__6] = lambda { T__6 = 6 }
  lazy_constants[:T__5] = lambda { T__5 = 5 }
  lazy_constants[:T__4] = lambda { T__4 = 4 }
  
  Initialize_test_lexer_SIGNATURES = priorize_signatures([], [CharStream], [CharStream, RecognizerSharedState])
  def initialize_test_lexer(arg0 = NO_ARG, arg1 = NO_ARG)
    case select_signature([arg0, arg1], Initialize_test_lexer_SIGNATURES)
    when 0 # []
      initialize_lexer()
    when 1 # [CharStream]
      input = arg0
      initialize_test_lexer(input, RecognizerSharedState.new)
    when 2 # [CharStream, RecognizerSharedState]
      input = arg0
      state = arg1
      initialize_lexer(input, state)
    end
  end
  
  alias_method :initialize, :initialize_test_lexer
  
  def get_grammar_file_name
    return "Test__.g"
  end
  
  def m_t__4
    begin
      _type = T__4
      _channel = DEFAULT_TOKEN_CHANNEL
      match(Character.new(?A))
      self.state.attr_type = _type
      self.state.attr_channel = _channel
    ensure
    end
  end
  
  def m_t__5
    begin
      _type = T__5
      _channel = DEFAULT_TOKEN_CHANNEL
      match(Character.new(?B))
      self.state.attr_type = _type
      self.state.attr_channel = _channel
    ensure
    end
  end
  
  def m_t__6
    begin
      _type = T__6
      _channel = DEFAULT_TOKEN_CHANNEL
      match(Character.new(?C))
      self.state.attr_type = _type
      self.state.attr_channel = _channel
    ensure
    end
  end
  
  def m_tokens
    alt1 = 3
    case (self.input.la(1))
    when Character.new(?A)
      alt1 = 1
    when Character.new(?B)
      alt1 = 2
    when Character.new(?C)
      alt1 = 3
    else
      nvae = NoViableAltException.new("", 1, 0, self.input)
      raise nvae
    end
    case (alt1)
    when 1
      m_t__4
    when 2
      m_t__5
    when 3
      m_t__6
    end
  end
end

TestLexer.main($*) if $0 == __FILE__

File: samples/antlr/TestLexer.java
// $ANTLR 3.1.1 Test__.g 2008-11-22 15:50:27

import org.antlr.runtime.*;
import java.util.Stack;
import java.util.List;
import java.util.ArrayList;

public class TestLexer extends Lexer {
    public static final int EOF=-1;
    public static final int T__6=6;
    public static final int T__5=5;
    public static final int T__4=4;

    // delegates
    // delegators

    public TestLexer() {;} 
    public TestLexer(CharStream input) {
        this(input, new RecognizerSharedState());
    }
    public TestLexer(CharStream input, RecognizerSharedState state) {
        super(input,state);

    }
    public String getGrammarFileName() { return "Test__.g"; }

    // $ANTLR start "T__4"
    public final void mT__4() throws RecognitionException {
        try {
            int _type = T__4;
            int _channel = DEFAULT_TOKEN_CHANNEL;
            // Test__.g:3:6: ( 'A' )
            // Test__.g:3:8: 'A'
            {
            match('A'); 

            }

            state.type = _type;
            state.channel = _channel;
        }
        finally {
        }
    }
    // $ANTLR end "T__4"

    // $ANTLR start "T__5"
    public final void mT__5() throws RecognitionException {
        try {
            int _type = T__5;
            int _channel = DEFAULT_TOKEN_CHANNEL;
            // Test__.g:4:6: ( 'B' )
            // Test__.g:4:8: 'B'
            {
            match('B'); 

            }

            state.type = _type;
            state.channel = _channel;
        }
        finally {
        }
    }
    // $ANTLR end "T__5"

    // $ANTLR start "T__6"
    public final void mT__6() throws RecognitionException {
        try {
            int _type = T__6;
            int _channel = DEFAULT_TOKEN_CHANNEL;
            // Test__.g:5:6: ( 'C' )
            // Test__.g:5:8: 'C'
            {
            match('C'); 

            }

            state.type = _type;
            state.channel = _channel;
        }
        finally {
        }
    }
    // $ANTLR end "T__6"

    public void mTokens() throws RecognitionException {
        // Test__.g:1:8: ( T__4 | T__5 | T__6 )
        int alt1=3;
        switch ( input.LA(1) ) {
        case 'A':
            {
            alt1=1;
            }
            break;
        case 'B':
            {
            alt1=2;
            }
            break;
        case 'C':
            {
            alt1=3;
            }
            break;
        default:
            NoViableAltException nvae =
                new NoViableAltException("", 1, 0, input);

            throw nvae;
        }

        switch (alt1) {
            case 1 :
                // Test__.g:1:10: T__4
                {
                mT__4(); 

                }
                break;
            case 2 :
                // Test__.g:1:15: T__5
                {
                mT__5(); 

                }
                break;
            case 3 :
                // Test__.g:1:20: T__6
                {
                mT__6(); 

                }
                break;

        }

    }


 

}